tests/grids/test_mortar_grid.py

```diff
@@ -0,0 +1,1094 @@
+"""Tests of the mortar grids. Mainly focuses on mappings between mortar grids and
+surrounding grids.
+
+The module contains the following groups of tests:
+    - One set of tests for 2d domains, where the 2d grid is replaced.
+    - One set of tests for 3d domains, where the 2d and 1d grids are replaced.
+    - test_pickle_mortar_grid: Individual test to verify that MortarGrids can be
+      pickled.
+
+A further description is given for each of the groups of tests.
+
+"""
+import pickle
+from itertools import count
+from pathlib import Path
+
+import numpy as np
+import pytest
+import scipy.sparse as sps
+
+import porepy as pp
+
+
+"""Tests of replacement of the 2d subdomain in a 2d domain with a single fracture.
+
+Various perturbations of the grid are tested. The tests are based on the following
+logic:
+    1. A 2d grid is created, with a single fracture.
+    2. A new 2d grid is created, and possibly perturbed.
+    3. The new grid is inserted into the old grid.
+    4. The projections between the old and the new grid are fetched, and some simple
+        sanity checks are done (common for all tests).
+    5. Specific checks of the projections are done for each test, based on knowledge of
+       how the new grid was perturbed and what the expected result is. This involves
+       checking the projections against hard coded values.
+
+The tests are made up of the following functions:
+    - A set of functions to test indivdiual replacement operations.
+    - A helper function _create_2d_mdg, which creates a 2d domain with a single fracture.
+    - A helper function _replace_2d_grid_fetch_projections, which replaces the 2d grid
+        in the domain and fetches the projections between the old and the new grid.
+
+"""
+
+
+def test_2d_domain_replace_2d_grid_by_identical_copy():
+    """Copy the higher dimensional grid and replace. The mapping should stay the same."""
+    # Create a first md grid
+    mdg, sd_old, old_projection = _create_2d_mdg([1, 2])
```

Comment:
Are you sure we shouldn't fixturize and copy?

Reply:
I think the current approach is reasonable. I tried to use a copy (not fixture, but to the same effect), but gained nothing in speed, so any bottleneck is somewhere else. I can try to add it, but then it would be for the sake of using fixtures.